src/uu/ls/src/ls.rs

```diff
@@ -3295,33 +3320,88 @@ fn create_hyperlink(name: &str, path: &PathData) -> String {
 /// This because we need to check the previous value in case we don't need
 /// the reset
 struct StyleManager {
+    /// last style that is applied, if `None` that means reset is applied.
     current_style: Option<Style>,
+    /// `true` if the initial reset is applied
+    initial_reset_is_done: bool,
 }
 
 impl StyleManager {
     fn new() -> Self {
         Self {
+            initial_reset_is_done: false,
             current_style: None,
         }
     }
 
-    fn apply_style(&mut self, new_style: &Style, name: &str) -> String {
-        if let Some(current) = &self.current_style {
-            if *current == *new_style {
-                // Current style is the same as new style, apply without reset.
-                let mut style = new_style.to_nu_ansi_term_style();
-                style.prefix_with_reset = false;
-                return style.paint(name).to_string();
+    fn apply_style(
```

Comment:
not sure it makes sense but maybe we should move the color functions into 
src/uu/ls/src/colors.rs (or something else)